SpinCore_pp/run_field_dep_justMw.py

```diff
@@ -88,162 +76,126 @@ def verifyParams():
-                            ('pulse_TTL',p90,'ph1',r_[0,1,2,3]),
-                            ('delay',tau),
-                            ('delay_TTL',deblank),
-                            ('pulse_TTL',2.0*p90,0),
-                            ('delay',deadtime),
-                            ('acquire',acq_time),
-                            ('delay',repetition),
-                            ('jumpto','start')
-                            ])
-                        #{{{
-                    if not phase_cycling:
-                        SpinCore_pp.load([
-                            ('marker','start',nScans),
-                            ('phase_reset',1),
-                            ('delay_TTL',deblank),
-                            ('pulse_TTL',p90,0.0),
-                            ('delay',tau),
-                            ('delay_TTL',deblank),
-                            ('pulse_TTL',2.0*p90,0.0),
-                            ('delay',deadtime),
-                            ('acquire',acq_time),
-                            ('delay',repetition),
-                            ('jumpto','start')
-                            ])
-                        #}}}
-                    print("\nSTOPPING PROG BOARD...\n")
-                    SpinCore_pp.stop_ppg();
-                    print("\nRUNNING BOARD...\n")
-                    SpinCore_pp.runBoard();
-                    raw_data = SpinCore_pp.getData(data_length, nPoints, nEchoes, nPhaseSteps, output_name)
-                    raw_data.astype(float)
-                    data_array = []
-                    data_array[::] = complex128(raw_data[0::2]+1j*raw_data[1::2])
-                    print("COMPLEX DATA ARRAY LENGTH:",shape(data_array)[0])
-                    print("RAW DATA ARRAY LENGTH:",shape(raw_data)[0])
-                    dataPoints = float(np.shape(data_array)[0])
-                    if x == 0 and B0_index == 0:
-                        time_axis = np.linspace(0.0,nEchoes*nPhaseSteps*acq_time*1e-3,dataPoints)
-                        data = ndshape([len(data_array),nScans,len(field_axis),1],['t','nScans','Field','power']).alloc(dtype=np.complex128)
-                        data.setaxis('t',time_axis).set_units('t','s')
-                        data.setaxis('nScans',r_[0:nScans])
-                        data.setaxis('Field',field_axis)
-                        data.setaxis('power',r_[powers])
-                        data.name(node_name)
-                        data.set_prop('acq_params',acq_params)
-                    data['nScans',x]['Field',B0_index]['power',0] = data_array
-                last_power = this_power
-            SpinCore_pp.stopBoard();
+total_pts = nPoints*nPhaseSteps
+assert total_pts < 2**14, "You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384"%total_pts
+#{{{ppg
+def run_sweep(fieldaxis, nScans=1, B0_index, data = None):
```

Comment:
why is this called run sweep?  This is not a sweep

Reply:
(this might be embedded in a loop that is a sweep, but this function seems to be just a 1-off pulse sequence.

Reply:
oops! changed name to run_scans